File: backend/src/main/java/com/viniciusfinger/rsflood/dto/PersonDTO.java
package com.viniciusfinger.rsflood.dto;

import com.viniciusfinger.rsflood.enums.BiologicalSex;
import com.viniciusfinger.rsflood.enums.Schooling;
import com.viniciusfinger.rsflood.model.PersonRace;

import lombok.Data;
import java.time.LocalDate;

@Data
public class PersonDTO {
    private Long id;

    private String name;

    private String cpf;

    private LocalDate birthDate;

    private PersonRace race;

    private BiologicalSex biologicalSex;

    private String phone;

    private Boolean hasDocuments;

    //Colabora com a renda familiar
    private Boolean collaborateFamilyIncome;

    private String profession;

    private Schooling schooling;

    private Integer dependentsNumber;

    private boolean isPCD;
}

File: frontend/src/app/model/dto/person-dto.ts
import { BiologicalSex } from "../enum/biological-sex";
import { PersonRace } from "../enum/person-race";
import { Schooling } from "../enum/schooling";

export class PersonDTO {
    id: number;
    name: string;
    cpf: string;
    birthDate: Date;
    race: PersonRace;
    biologicalSex: BiologicalSex;
    phone: string;
    hasDocuments: boolean;
    collaborateFamilyIncome: boolean;
    profession: string;
    schooling: Schooling;
    dependentsNumber: number;
    isPCD: boolean;

    constructor(
        id: number,
        name: string,
        cpf: string,
        birthDate: Date,
        race: PersonRace,
        biologicalSex: BiologicalSex,
        phone: string,
        hasDocuments: boolean,
        collaborateFamilyIncome: boolean,
        profession: string,
        schooling: Schooling,
        dependentsNumber: number,
        isPCD: boolean
    ) {
        this.id = id;
        this.name = name;
        this.cpf = cpf;
        this.birthDate = birthDate;
        this.race = race;
        this.biologicalSex = biologicalSex;
        this.phone = phone;
        this.hasDocuments = hasDocuments;
        this.collaborateFamilyIncome = collaborateFamilyIncome;
        this.profession = profession;
        this.schooling = schooling;
        this.dependentsNumber = dependentsNumber;
        this.isPCD = isPCD;
    }
}
